bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -531,6 +531,26 @@ class BcStTest : public ::testing::Test {
   void onTimerImp() { stateTransfer_->onTimerImp(); }
   SourceSelector& getSourceSelector() { return stateTransfer_->sourceSelector_; }
 
+  bool compareSourceSelectorMetricKeyVal(const std::string& key, uint64_t val) {
```

Comment:
An alternative would be to have compareSourceSelectorMetricKeyVal -> assertSourceSelectorMetricsValue
This is more "google style", and clearer in my opinion, since if there is a failure you understand exactly why it fails and get a clear printout of the values.
In your code, when we fail we do not get clear information on the failure.

```
void assertSourceSelectorMetricsValue(const std::string& key, uint64_t val) {
   if (key == "total_replacements_") {
      ASSERT_EQ(getSourceSelector().metrics_.total_replacements_.Get().Get(), val));
    } else if (key == "replacement_due_to_periodic_change_") {
      ASSERT_EQ(getSourceSelector().metrics_.replacement_due_to_periodic_change_.Get().Get(), val);
    } else if (key == "replacement_due_to_source_same_as_primary_") {
      ASSERT_EQ(getSourceSelector().metrics_.replacement_due_to_source_same_as_primary_.Get().Get(), val);
    } else if (key == "replacement_due_to_retransmission_timeout_") {
      ASSERT_EQ(getSourceSelector().metrics_.replacement_due_to_retransmission_timeout_.Get().Get(), val);
    } else {
      FAIL() << "Unexpected key!";
    }
}
```

To call from outside:
`ASSERT_NO_FATAL_FAILURE(checkDefaultSourceSelectorMetricValues());`